app/scripts/ui.js

```diff
@@ -77,6 +86,100 @@ async function start() {
         } else {
           initializeUiWithTab(activeTab);
         }
+        await loadPhishingWarningPage();
+      }
+    };
+
+    /**
+     * An error thrown if the phishing warning page takes too long to load.
+     */
+    class PhishingWarningPageTimeoutError extends Error {
+      constructor() {
+        super('Timeout failed');
+      }
+    }
+
+    /**
+     * Load the phishing warning page temporarily to ensure the service
+     * worker has been registered, so that the warning page works offline.
+     */
+    loadPhishingWarningPage = async function () {
+      const currentPlatform = getPlatform();
+      let isSWAlreadyRegistered;
+
+      if (currentPlatform === PLATFORM_FIREFOX) {
+        isSWAlreadyRegistered = window.sessionStorage.getItem(
```

Comment:
`sessionStorage` is not a suitable substitute for `storage.session` here. It gets refreshed every page load, so it would always be empty.

Reply:
I found a bug tracking this for Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1687778

Maybe for now we can just.... print a console error and link this bug in a comment, since we're not actively working on MV3 support in Firefox right now.

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/16029/commits/c213777d4eee177548791e77227a68df0452c87f)